src/main/scala/amba/axi4/Deinterleaver.scala

```diff
@@ -27,28 +28,39 @@ class AXI4Deinterleaver(maxReadBytes: Int)(implicit p: Parameters) extends LazyM
       val beats = (maxReadBytes+beatBytes-1) / beatBytes
 
       // This adapter leaves the control + write paths completely untouched
-      out.ar <> in.ar
-      out.aw <> in.aw
-      out.w <> in.w
-      in.b <> out.b
+      out.ar :<> in.ar
+      out.aw :<> in.aw
+      out.w :<> in.w
+      in.b :<> out.b
 
       if (beats <= 1) {
         // Nothing to do if only single-beat R
-        in.r <> out.r
+        in.r :<> out.r
       } else {
         // Queues to buffer R responses
         val qs = Seq.tabulate(endId) { i =>
           val depth = edgeOut.master.masters.find(_.id.contains(i)).flatMap(_.maxFlight).getOrElse(0)
           if (depth > 0) {
-            Module(new Queue(out.r.bits.cloneType, beats)).io
+            val q = Module(new Queue(out.r.bits.cloneType, beats))
+            q.suggestName(s"queue_${i}")
+            q.io
           } else {
-            Wire(new QueueIO(out.r.bits.cloneType, beats))
+            // These are unused IDs and should be never used.
+            // But, to satisfy type checks we must produce a Wire of the
+            // correct type.
+            val q = Wire(new QueueIO(out.r.bits.cloneType, beats))
+            q.suggestName(s"queue_wire_${i}")
+            assert(!q.enq.valid, s"ID ${i} should not be used")
+            // These could just be marked DontCare
+            q.deq :<> q.enq
```

Comment:
@terpstra can you re-review this bit? This is all that is needed to be DontCare, not every queue, but I wired them up to eachother and added an assertion instead

Reply:
Ahh. This makes sense. Thanks for figuring it out! I do not think cross connecting them is wise, however. Use a DontCare instead.

Reply:
done